src/clib/pio_internal.h

```diff
@@ -152,6 +152,9 @@ extern "C" {
     int compute_counts(iosystem_desc_t *ios, io_desc_t *iodesc, int maplen,
                        const int *dest_ioproc, const PIO_Offset *dest_ioindex,
                        MPI_Comm mycomm);
+
+    /* Compute mapping for subset rearranger. */
+    int default_subset_partition(iosystem_desc_t *ios, io_desc_t *iodesc);    
```

Comment:
The as yet unrealized vision here is that the user would be able to supply an alternative subset partitioning function.  Requirements of this function are that there be exactly one io task per compute task group.